jparest/src/test/resources/data-h2.sql

```diff
@@ -22,9 +22,9 @@ INSERT INTO dummy_EntityB (id, tenant_id) VALUES
     ('b7e813a2-bb28-11ec-8422-0242ac120009', 'b7e813a2-bb28-11ec-8422-0242ac120002'),
     ('b7e813a2-bb28-11ec-8422-0242ac120010', 'b7e813a2-bb28-11ec-8422-0242ac120002');
 
-INSERT INTO dummy_EntityC (id, tenant_id, description, index) VALUES
-    ('b7e813a2-bb28-11ec-8422-0242ac130001', 'b7e813a2-bb28-11ec-8422-0242ac120002', 'Dummy Entity C number 1' , 1),
-    ('b7e813a2-bb28-11ec-8422-0242ac130002', 'b7e813a2-bb28-11ec-8422-0242ac120002', 'Dummy Entity C number 2', 2);
+INSERT INTO dummy_EntityC (id, tenant_id, description, index, dob, instant) VALUES
+    ('b7e813a2-bb28-11ec-8422-0242ac130001', 'b7e813a2-bb28-11ec-8422-0242ac120002', 'Dummy Entity C number 1' , 1, '2012-12-12', '2012-12-12T10:00:00.000+00:00'),
```

Comment:
I think we should also add tests to deal with timezones that are not just UTC. See early comment about if we should use instant. In the future we may need to work with different timezones more so we can't loose the original timezone offset information. For example, if recording time overseas when we need to attribute accruals to a given day it needs to use the timezone from the overseas territory.

Reply:
I have changed one BDD test to post an `Instant` value with offset +02:00: `1989-05-21T10:00:00.000+02:00`, the value returned is `1989-05-21T08:00:00.000+00:00`. So it is technically the same instant, but it lost the original offset information (see `jpaRest.feature`, line 124). This is due to the fact that the `timestamp with time zone` sql type is always stored as UTC, and this is regardless of the java type (Instant or ZonedDateTime). As a consequence, the original time zone offset/ID (e.g. `+02:00`/`Europe/Paris`) is lost when the value is read again.

If we are to support time zone IDs (e.g. `Europe/Paris`), then `ZonedDateTime` would be the best fit. However, it's not supported natively by JPA/Hibernate, so we would have to come up with our own solution. This would most probably involve an additional column to store the Time Zone ID.

In all cases, I think this is beyond the scope of current task and should be progressed as a separate technical ticket. If you agree, I can create a JIRA ticket to capture the problem.
 
P.S. Excerpt from [Postgres Documentation](https://www.postgresql.org/docs/current/datatype-datetime.html) : 
"For timestamp with time zone, the internally stored value is always in UTC (Universal Coordinated Time, traditionally known as Greenwich Mean Time, GMT). An input value that has an explicit time zone specified is converted to UTC using the appropriate offset for that time zone"

Reply:
I agree that we should split this out to another ticket. Please create a task with the `Technical` label you found and we can prioritise that away from this PR

Reply:
Ticket created: https://collaboration.homeoffice.gov.uk/jira/browse/EAHW-2628